tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -409,7 +409,7 @@ def _node_tag(self, tag, context, ros_config, default_machine, is_test=False, ve
                 elif tag_name == 'env':
                     self._env_tag(t, env_context, ros_config)
                 else:
-                    ros_config.add_config_error("WARN: unrecognized '%s' tag in <node> tag. Node xml is %s"%(t.tagName, tag.toxml()))
+                    ros_config.add_config_error("WARN: unrecognized '%s' tag in <node> or <test> tag. Node xml is %s"%(t.tagName, tag.toxml()))
```

Comment:
Since `tag` contains the actual parent of `t` couldn't you extract the exact tagname from it?